### ISSUE TITLE
[Fix] [forecasting] test score is quite low

### ISSUE THREAD

**Issue by lisphilar** [lisphilar/covid19-sir#783]
## Summary
Test score of regression (currently only with Elastic Net) is quite low for Italy case (around 0, R-squared).

## Codes

```Python
import covsirphy as cs
# Dataset preparation
data_loader = cs.DataLoader("input")
jhu_data = data_loader.jhu()
oxcgr_data = data_loader.oxcgrt()
# Scenario analysis
snl = cs.Scenario(country="Italy")
snl.register(jhu_data, extras=[oxcgrt_data]))
snl.trend()
snl.estimate(cs.SIRF)
fit_dict = snl.fit()
print(fit_dict["score_test"])
```

## Outputs
` -0.004088981878122677`

## Environment

- CovsirPhy version: 2.20.1
- Python version: 3.9.5
- Installation: poetry
- System: WSL (Ubuntu)

**Comment by lisphilar** [comment]
Note that low accuracy of forecasting (i.e. predicted number of cases is far from actual values) will be the next issue.

**Comment by lisphilar** [comment]
The first strategy:
- PCA for dimensionality reduction to remove un-nenecessary information
- Decision Tree Regressor: requires little data preparation and rubust for outliers
- Max depth of tree will be determined by grid search

https://dorukkilitcioglu.com/2018/08/11/pca-decision-tree.html
https://scikit-learn.org/stable/modules/tree.html#tree

**Comment by lisphilar** [comment]
Only when test score of PCA + Decision Tree Regressor is higher than Elastic Net, PCA + Decision Tree Regressor will be used.

**Comment by lisphilar** [comment]
This is related to #778.

**Comment by lisphilar** [comment]
With #785, around 0.98 for R-squared.
